### DISCUSSION TITLE
Welcome to discuss or ask questions about Openwrt!

### DISCUSSION THREAD

**Discussion by nicholas-opensource** [nicholas-opensource/OpenWrt-Autobuild#1] [General]
## 👋 Welcome!
You will have a different experience from the telegram group, anyone can feel free to discuss or ask questions about OpenWrt! You won't get despised here unless you are not friendly.

Here you can discuss not only this repo but also any repositories you like~ Just provide the source code address~

I will reply to your comments one by one as soon as possible!

**Comment by nicholas-opensource** [comment]
Oh, btw, you need to comply with open source agreements~

**Comment by nicholas-opensource** [comment]
中文提问也是可以的哦~

**Comment by Kurokosama** [comment]
Hello.

**Reply by nicholassun** [reply]
Hello there

**Comment by ghost** [comment]
大神，你的固件自己装 OpenClash 始终遇到问题，但是我又喜欢这个最极简的固件。当前版本是 21.02-SNAPSHOT r15894-69b77dc3b4，OpenClash v0.40.7 能启动但是上不了网，v0.40.8 以上版本启动不了。

**Reply by nicholas-opensource** [reply]
请问您有日志提供吗，自己安装OpenClash是非常不错的选择

**Reply by ghost** [reply]
感谢您的回复！我去开一个新贴贴日志

**Comment by omransh** [comment]
I want to ask a question.
I downloaded openwrt source code, and I want to modify the iptables,NAT,filter files , then to compile the source code.
I face a problem which it is that I didn't find the path of these files for example in target or package or any where.

So, if you can tell me how can I find the path of these files, and If you give me some notes.

Thanks.

**Reply by nicholas-opensource** [reply]
This might help you

https://openwrt.org/docs/guide-user/firewall/netfilter_iptables/netfilter_openwrt

**Comment by omransh** [comment]
I want to ask you a question.

Is there in openwrt source code a file responsible for packet capture or sniffer or raw socket?

Firewall can make reject, accept or redirect, but how it can know the packet detailed, on what it depend on which file ?

**Reply by nicholas-opensource** [reply]
Tcp dump

https://openwrt.org/packages/pkgdata/tcpdump

**Reply by omransh** [reply]
Is tcp dump already installed and worked in openwrt ? or I should make something to activate its role ?

**Reply by nicholas-opensource** [reply]
```
opkg update && opkg install tcpdump
```
You can use like `tcpdump -i any -w files.cap` to export cap files and open it in the Wireshark

**Comment by omransh** [comment]
Brother, 
I search about .c file written in c language that responsible for packet capture(like packet socket), and used for nating,filtering ..etc .
So, can you tell me where can I find this file ?

**Comment by omransh** [comment]
I need to see the code of packet capture(udp,tcp) in networking in the Linux Kernel.
Can you tell the location of file that contain the code?

**Reply by nicholas-opensource** [reply]
I am not very familiar with the Linux network stack. I'm sorry I can't help you

**Comment by omransh** [comment]
Hello,
After compilation source code, a build_dir file appeared.
Q1: Can you tell me what is the responsible file for creation it ?
Q2: Can I edit build_dir ?

**Comment by Alexrobot03** [comment]
集成大鹅了吗～哥哥

**Reply by nicholas-opensource** [reply]
当然，路径/usr/bin/dae，daed也集成了

**Reply by Alexrobot03** [reply]
哥哥，TP 5480咋样

**Reply by nicholas-opensource** [reply]
初始版本的配置是很好的，现在有些减配，而且方案现在有些老了。此外系统比较难用，需要一点设置技巧。高通的配置对全安卓方案来说很不错，但是对于使用bcm的Apple设备来说不太理想，没有针对awdl进行优化。

**Comment by Alexrobot03** [comment]
N100刷完进不去系统，直接进了BIOS，404的没问题，不知道咋回事～

**Reply by nicholas-opensource** [reply]
俺这是legacy

**Reply by Alexrobot03** [reply]
我这个BIOS好像没有这个模式，尴尬> 俺这是legacy

**Reply by nicholas-opensource** [reply]
行吧，为了您，加上combine-efi吧

**Reply by Alexrobot03** [reply]
> 行吧，为了您，加上combine-efi吧

那怎么好意思，嘻嘻～

**Reply by nicholas-opensource** [reply]
https://github.com/nicholas-opensource/OpenWrt-Autobuild/actions/runs/5989481675
跑完了，自取吧

**Reply by Alexrobot03** [reply]
取到了，BOT定制版，真香💋💋💋💋💋💋💋💋💋

**Reply by Alexrobot03** [reply]
爱您💋💋💋💋💋

**Reply by nicholas-opensource** [reply]
以后就是常规发布了

**Reply by Alexrobot03** [reply]
好像opkg update完了后没有ssrp和openclash这些玩意儿了～> 以后就是常规发布了

**Reply by Alexrobot03** [reply]
好像opkg update完了后没有ssrp和openclash这些玩意儿了～> 以后就是常规发布了

**Reply by nicholas-opensource** [reply]
本来openwrt就没有啊，ssrp早就去掉了，因为firewall3太老了。现在有dae谁还用这种软件）

**Reply by Alexrobot03** [reply]
dae用起来莫名的丝滑，应该不是错觉～

**Reply by nicholas-opensource** [reply]
当然不是的～

**Reply by Alexrobot03** [reply]
您的头像那嘴唇看起来好sexy，有点骚😆

**Reply by nicholas-opensource** [reply]
歪，110吗？

**Reply by Alexrobot03** [reply]
大佬帮忙算个命，遇到一个很奇葩的菜鸡问题，手机是水果，WiFi是5480，不开代理10000.gd.cn手机电信本地测速650Mbps（开dae速度差不多），但是用openclash测速10000.gd.cn手机端速度850Mbps，这是什么原理。

**Reply by nicholas-opensource** [reply]
建议用ookla speedtest，电信测速我无力吐槽
